contracts/instruments/RibbonThetaVaultYearn.sol

```diff
@@ -0,0 +1,816 @@
+        )
+    {
+        uint256 total = lockedAmount.add(currentAssetBalance);
+
+        uint256 shareSupply = totalSupply();
+
+        // solhint-disable-next-line
+        // Following the pool share calculation from Alpha Homora: https://github.com/AlphaFinanceLab/alphahomora/blob/340653c8ac1e9b4f23d5b81e61307bf7d02a26e8/contracts/5/Bank.sol#L111
+        withdrawAmount = share.mul(total).div(shareSupply);
+        newAssetBalance = total.sub(withdrawAmount);
+        newShareSupply = shareSupply.sub(share);
+    }
+
+    /**
+     * @notice Returns the max withdrawable shares for all users in the vault
+     */
+    function maxWithdrawableShares() public view returns (uint256) {
+        uint256 withdrawableBalance = assetBalance();
+        uint256 total = lockedAmount.add(withdrawableBalance);
+        return
+            withdrawableBalance
+                .mul(totalSupply())
+                .div(total)
+                .sub(MINIMUM_SUPPLY)
+                .sub(queuedWithdrawShares);
+    }
+
+    /**
+     * @notice Returns the max amount withdrawable by an account using the account's vault share balance
+     * @param account is the address of the vault share holder
+     * @return amount of `asset` withdrawable from vault, with fees accounted
+     */
+    function maxWithdrawAmount(address account)
+        external
+        view
+        returns (uint256)
+    {
+        uint256 maxShares = maxWithdrawableShares();
+        uint256 share = balanceOf(account);
+        uint256 numShares = min(maxShares, share);
+
+        (uint256 withdrawAmount, , ) =
+            _withdrawAmountWithShares(numShares, assetBalance());
+
+        return withdrawAmount;
+    }
+
+    /**
+     * @notice Returns the number of shares for a given `assetAmount`.
+     *         Used by the frontend to calculate withdraw amounts.
+     * @param assetAmount is the asset amount to be withdrawn
+     * @return share amount
+     */
+    function assetAmountToShares(uint256 assetAmount)
+        external
+        view
+        returns (uint256)
+    {
+        uint256 total = lockedAmount.add(assetBalance());
+        return assetAmount.mul(totalSupply()).div(total);
+    }
+
+    /**
+     * @notice Returns an account's balance on the vault
+     * @param account is the address of the user
+     * @return vault balance of the user
+     */
+    function accountVaultBalance(address account)
+        external
+        view
+        returns (uint256)
+    {
+        (uint256 withdrawAmount, , ) =
+            _withdrawAmountWithShares(balanceOf(account), assetBalance());
+        return withdrawAmount;
+    }
+
+    /**
+     * @notice Returns the vault's total balance, including the amounts locked into a short position
+     * @return total balance of the vault, including the amounts locked in third party protocols
+     */
+    function totalBalance() public view returns (uint256) {
+        return lockedAmount.add(assetBalance());
+    }
+
```

Comment:
Spoke to the yearn team, one getter that would be helpful to add is `yieldTokenBalance` (or if you can think of a better name). The getter is used to show how much yvault tokens are held by the vault for purposes of attributing profit shares for us.

It would look like this:
```solidity
return IERC20(address(collateralToken)).balanceOf(address(this)).add(lockedAmount);
```

Reply:
`collateralTokenBalance`

Reply:
addressed